site/src/pages/TaskPage/TaskPage.tsx

```diff
@@ -465,6 +525,64 @@ const TaskAppIFrame: FC<TaskAppIFrameProps> = ({ task, app, active }) => {
 	);
 };
 
+type TaskStatusLinkProps = {
+	uri: string;
+};
+
+const TaskStatusLink: FC<TaskStatusLinkProps> = ({ uri }) => {
+	const linkFormat = getLinkFormat(uri);
+
+	return (
+		<Button asChild variant="outline" size="sm" className="min-w-0">
+			<a href={linkFormat.href} target="_blank" rel="noreferrer">
+				{linkFormat.icon}
+				{linkFormat.label}
+			</a>
+		</Button>
+	);
+};
+
+type LinkFormat = {
+	icon: ReactNode;
+	label: string;
+	href: string;
+};
+
+const getLinkFormat = (uri: string): LinkFormat => {
```

Comment:
Could we split this into 2 functions? `parseUri` that returns either an issue, a PR, or a generic link object, and `renderLink` that accepts the result of the former as a param and returns a React node.